### PR TITLE
Add reference for audit API group

### PR DESCRIPTION
**Pull request by tengqm**
This is a reference for "audit.k8s.io/v1" API group generated from kubernetes-sigs/reference-docs/genref tool.
More specifically, it is generated using the following command:

```shell
./genref -include apiserver-audit
```

### CODE REVIEW

**Comment by tengqm** [comment]
Preview: https://deploy-preview-26606--kubernetes-io-master-staging.netlify.app/docs/reference/command-line-tools-reference/apiserver-audit.v1/

**Comment by kbhawkey** [comment]
Hi @tengqm . Here is initial feedback on the rendered output.
- Some fields have an asterisk? The field type is listed below the field name. Would it make sense to list
  the type next to the field name?
- The name of the resource type is level H3? Could remove the top TOC (Resource types) and use level H2 for each type?
- If fields are missing a description, should you substitute some common text into the blank table fields?
- The file is named, `docs/reference/command-line-tools-reference/apiserver-audit.v1/` and the title of the page is:
   audit.k8s.io/v1? Should the title include "apiserver"?
- You may want to create a new directory in the left navigation tree to house the configuration references? Also, one of the link locations is: `/docs/reference/command-line-tools-reference/apiserver-audit.v1/#audit-k8s-io-v1-Stage? Do you want to replace the "dot" in the filename to `apiserver-audit-v1`?
- Do you like the table rows with "shading/stripes"?

**Comment by tengqm** [comment]
> * Some fields have an asterisk?

A blue star indicates that field is NOT marked as optional.

> The field type is listed below the field name. Would it make sense to list the type next to the field name?

My overall observation is that in most if not all configuration structs, the field names and field type names tend to be long. The descriptions for field member are markdown paragraphs. Which are long too. To make sure the whole table is always readable on all devices, wrapping the type of the field to a second line is a good choice.

> * The name of the resource type is level H3? Could remove the top TOC (Resource types) and use level H2 for each type?

Yes. That is doable. Will revise.

> * If fields are missing a description, should you substitute some common text into the blank table fields?

That makes sense.

> * The file is named, `docs/reference/command-line-tools-reference/apiserver-audit.v1/` and the title of the page is:  audit.k8s.io/v1? Should the title include "apiserver"?

The `apiserver-audit` name comes from the `config.yaml` file. It is just a unique identifier for the API package. The `apiserver` part comes from the package path. `audit.k8s.io/v1` comes from the source code parsing. Maybe I can go through the config file and rename the identifier accordingly.

> * You may want to create a new directory in the left navigation tree to house the configuration references? Also, one of the link locations is: `/docs/reference/command-line-tools-reference/apiserver-audit.v1/#audit-k8s-io-v1-Stage? Do you want to replace the "dot" in the filename to `apiserver-audit-v1`?

I don't see that renaming an improvement. It will make version parsing difficult given that group names for these config APIs vary in length. Keeping a dot in the name helps extract/identify version string from the file name.

> * Do you like the table rows with "shading/stripes"?

I can remove the `table-striped` class. Let's see if such a change helps improve the look & feel.

Thanks for the detailed review.

**Comment by tengqm** [comment]
Preview link: https://deploy-preview-26606--kubernetes-io-master-staging.netlify.app/docs/reference/config-api/apiserver-audit.v1/

**Comment by tengqm** [comment]
/hold
Depends on: https://github.com/kubernetes-sigs/reference-docs/pull/214

**Comment by tengqm** [comment]
/hold cancel
kubernetes-sigs/reference-docs#214 is in.

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 1]
```diff
@@ -46,7 +46,8 @@ Each request can be recorded with an associated _stage_. The defined stages are:
 - `Panic` - Events generated when a panic occurred.
 
 {{< note >}}
-Audit events are different from the
+An audit [Event](/docs/reference/config-api/apiserver-audit.v1/#audit-k8s-io-v1-Event)
```
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/audit.md:49]: nit: You may want to update this link text to something more meaningful.
Something like:  "Audit Event configuration"?

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 2]
```diff
@@ -95,6 +96,9 @@ If you're crafting your own audit profile, you can use the audit profile for Goo
 [configure-helper.sh](https://github.com/kubernetes/kubernetes/blob/{{< param "githubbranch" >}}/cluster/gce/gci/configure-helper.sh)
 script, which generates an audit policy file. You can see most of the audit policy file by looking directly at the script.
 
+You can also refer to the [`Policy` reference](/docs/reference/config-api/apiserver-audit.v1/#audit-k8s-io-v1-Policy)
```
Comment (kbhawkey) [inline content/en/docs/tasks/debug-application-cluster/audit.md:99]: nit: possibly:
You can also refer to the [`Policy` configuration reference](...) ...

File: content/en/docs/tasks/debug-application-cluster/audit.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/tasks/debug-application-cluster/audit.md:49]: Done.

File: content/en/docs/reference/_index.md [hunk 3]
```diff
@@ -63,6 +67,15 @@ client libraries:
 * [Scheduler Profiles](/docs/reference/scheduling/config#profiles)
 
 
+## Config APIs
+
+This section hosts the documentation for "unpublished" APIs which are used to
+configure the behavior for kubernetes components or tools. Most of these APIs
```
Comment (kbhawkey) [inline content/en/docs/reference/_index.md:73]: nit: This section hosts the documentation for "unpublished" APIs which are used to
configure Kubernetes components or tools.

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-26606--kubernetes-io-master-staging.netlify.app/docs/reference/#config-apis

File: content/en/docs/reference/_index.md [hunk 4]
```diff
@@ -63,6 +67,15 @@ client libraries:
...
+are not exposed by the API server in a RESTful way though they are essential
+for a user or an operator to use or manage a cluster.
+
+* [the `audit.k8s.io/v1` API](/docs/reference/config-api/apiserver-audit.v1/)
```
Comment (kbhawkey) [inline content/en/docs/reference/_index.md:77]: nit: Could you remove "the"?

Comment (tengqm) [reply content/en/docs/reference/_index.md:77]: Done

File: content/en/docs/reference/_index.md (hunk 3 above)
Comment (tengqm) [reply content/en/docs/reference/_index.md:73]: Done.

**Comment by kbhawkey** [comment]
The Audit configuration page publishes under `reference/config-api` but will not
display under this directory until the section _index file merges (related PR).
/lgtm

File: content/en/docs/tasks/debug-application-cluster/audit.md [hunk 5]
```diff
@@ -250,3 +251,5 @@ By default truncate is disabled in both `webhook` and `log`, a cluster administr
 ## {{% heading "whatsnext" %}}
 
 * Learn about [Mutating webhook auditing annotations](/docs/reference/access-authn-authz/extensible-admission-controllers/#mutating-webhook-auditing-annotations).
+* Read the [reference for `audit.k8s.io` API group](/docs/reference/config-api/apiserver-audit.v1/).
```
Comment (sftim) [inline content/en/docs/tasks/debug-application-cluster/audit.md:254]: nit: we could actually link directly to the Event and Policy definitions.

**Review by sftim — COMMENTED** [review]
/lgtm

**Comment by sftim** [comment]
Taking https://github.com/kubernetes/website/pull/26606#issuecomment-808750450 as LGTM by another party
/approve

File: content/en/docs/tasks/debug-application-cluster/audit.md (hunk 5 above)
Comment (tengqm) [reply content/en/docs/tasks/debug-application-cluster/audit.md:254]: good idea, will do.

Comment (tengqm) [reply content/en/docs/tasks/debug-application-cluster/audit.md:254]: good idea, will do.

**Pull request merged**